examples/ensemble/plot_gradient_boosting_quantile.py

```diff
@@ -286,15 +286,12 @@ def coverage_fraction(y, y_low, y_high):
 pprint(search_95p.best_params_)
 
 # %%
-# This time, shallower trees are selected and lead to a more constant piecewise
-# and therefore more robust estimation of the 95th percentile. This is
-# beneficial as it avoids overfitting the large outliers of the log-normal
-# additive noise.
-#
-# We can confirm this intuition by displaying the predicted 90% confidence
-# interval comprised by the predictions of those two tuned quantile regressors:
-# the prediction of the upper 95th percentile has a much coarser shape than the
-# prediction of the lower 5th percentile:
+# The result shows that the hyper-parameters for the 95th percentile regressor
+# identified by the grid search are roughly in the same range as the hand-
+# tuned hyper-parameters for the median regressor and the hyper-parameters
+# identified by the grid search for the 5th percentile regressor. However, the
+# hyper-parameter grid searches did lead to an improved 90% confidence
+# interval which can be seen below:
```

Comment:
This change also looses some, in my opinion, valuable information.

Reply:
I think the issue here is that the original hyperparameter space didn't include 0.2 learning rate, and therefore the statement was true, and expanding on the learning rate space, makes the example faster, and also makes it actually choose shallow trees in the first place. So I'm not sure if the information provided was kinda true in the first place.

Or do you mean something else @lorentzenchr ?

Reply:
In the last plot, it's visible that the lower (5%) quantile is much more fine grained than the upper (95%) quantile. This statement, however, is less obvious from the new tuned parameters. I would at least comment on the plot.

Reply:
Regarding the tree depth: yes, based on my results deeper trees are actually not needed to achieve a good fit for the 5th percentile regressor. This is why I changed the documentation. 
Regarding the granularity of the 5th and 95th percentile: true, with my changes the documentation on this gets lost. I'll change this.

Reply:
I'll also try using `HalvingRandomSearchCV` as suggested and will report back my results.

Reply:
@marenwestermann Do you need any help or just more time?

Reply:
@lorentzenchr Thank you for checking in! I was moving house in the meantime. :) I addressed the comments, let me know if you would like more changes.